asv_benchmarks/benchmarks/benchmarks.py

```diff
@@ -0,0 +1,103 @@
+# Write the benchmarking functions here.
+# See "Writing benchmarks" in the airspeed velocity docs for more information.
+# https://asv.readthedocs.io/en/stable/
+import numpy as np
+from scipy.stats import circstd
+import arviz as az
+from arviz.stats.stats_utils import _circular_standard_deviation, stats_variance_2d
+from arviz.numeric_utils import _fast_kde, _fast_kde_2d, histogram
+
+
+class Hist:
+    params = (True, False)
+    param_names = ("Numba",)
+
+    def setup(self, numba_flag):
+        self.data = np.random.rand(10000, 1000)
+        if numba_flag:
+            az.Numba.enable_numba()
+        else:
+            az.Numba.disable_numba()
+
+    def time_histogram(self, numba_flag):
+        histogram(self.data, bins=100)
+
+
+class Variance:
+    params = (True, False)
+    param_names = ("Numba",)
+
+    def setup(self, numba_flag):
+        self.data = np.random.rand(10000, 1000)
+        if numba_flag:
+            az.Numba.enable_numba()
+        else:
+            az.Numba.disable_numba()
+
+    def time_variance(self, numba_flag):
+        stats_variance_2d(self.data)
+
+
+class CircStd:
+    params = (True, False)
+    param_names = ("Numba",)
+
+    def setup(self, numba_flag):
+        self.data = np.random.randn(10000, 1000)
+        if numba_flag:
+            self.circstd = _circular_standard_deviation
+        else:
+            self.circstd = circstd
+
+    def time_circ_std(self, numba_flag):
+        self.circstd(self.data)
+
+
+class Fast_Kde_1d:
+    params = [(True, False), (10**5, 10**6, 10**7)]
+    param_names = ("Numba", "n")
+
+    def setup(self, numba_flag, n):
+        self.x = np.random.randn(n//10, 10)
+        if numba_flag:
+            az.Numba.enable_numba()
+        else:
+            az.Numba.disable_numba()
+
+    def time_fast_kde_normal(self, numba_flag, n):
+        _fast_kde(self.x)
+
+class Fast_KDE_2d:
+    params = [(True, False), (10**5, 10**6)]
```

Comment:
I have realized that with these shapes, there is no speedup when using numba, I tried to look if the same issue as with fast_kde_1d happened where the speedup was lost in a future PR, and it looks like the speed-up is still there, but is only seen for larger sizes. 

Could we try something like this and see how do dimensions affect performance?

```
class Fast_KDE_2d:
    params = [
        (True, False), 
        ((100, 10**4), (10**4, 100), (1000, 1000))
    ]
    param_names = ("Numba", "shape")

    def setup(self, numba_flag, shape):
        self.x = np.random.randn(*shape)
        self.y = np.random.randn(n//10, 10)
        if numba_flag:
        ...
```

Reply:
Ooh, I'll implement this.

Also, in #1088 you mentioned that more functions need to be covered in benchmarks and later on memory benchmarks and line profiling also to be added, so should I start adding all this now in this PR or sometime later?

Reply:
Maybe it's better to wait for another PR

Reply:
> I have realized that with these shapes, there is no speedup when using numba, I tried to look if the same issue as with fast_kde_1d happened where the speedup was lost in a future PR, and it looks like the speed-up is still there, but is only seen for larger sizes.
> 
> Could we try something like this and see how do dimensions affect performance?
> 
> ```
> class Fast_KDE_2d:
>     params = [
>         (True, False), 
>         ((100, 10**4), (10**4, 100), (1000, 1000))
>     ]
>     param_names = ("Numba", "shape")
> 
>     def setup(self, numba_flag, shape):
>         self.x = np.random.randn(*shape)
>         self.y = np.random.randn(n//10, 10)
>         if numba_flag:
>         ...
> ```

It gives the following error:
```
ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 1000000 and the array at index 1 has size 100
```
Do `x` and `y` have to be of the same size always?

Also, I did `self.y = np.random.randn(shape[0]//10, 10)` instead of `self.y = np.random.randn(n//10, 10)`. This is what you meant?

Reply:
I forgot to update y, it should be created like x, they must be the same shape. Also, there shoud be no n instances anymore

Reply:
Yup, I had removed `n`. The only problem was the different sizes of `x` and `y`. You'll see after the tests finish that there's still no speed-up :(